engines/dreamweb/stubs.cpp

```diff
@@ -963,6 +964,14 @@ void DreamWebEngine::useTimedText() {
 
 	const uint8 *string = (const uint8 *)_timedTemp._string;
 	printDirect(string, _timedTemp._x, _timedTemp._y, 237, true);
+	const char *theText = (const char *)string;
+	if (_lastText != theText) {
```

Comment:
Could `_ttsMan->isSpeaking()` be used instead of checking if the string changed?

Reply:
The short answer is no.

By default `TextToSpeechManager::say()` already does nothing when trying to speak again the same string that it is currently speaking. But the issue here is that we also want to avoid repeating again and again the same string. Without this check, as soon has it has finished speaking the text, it will start speaking it again because `useTimedText()` gets called again and again (multiple times per second). We only want to speak the text when a new text is displayed.

Reply:
Fair enough :) I'm just not a fan of keeping a state of the currently playing sentence